spring-cloud-starter-stream-processor-tasklaunchrequest-transform/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/transform/processor/TasklaunchrequestTransformProcessorConfiguration.java

```diff
@@ -99,7 +99,7 @@ public Object setupRequest(Message<?> message) throws Exception{
 				deploymentProperties,
 				applicationName);
 		Map<String, Object> headerMap = new HashMap<>(message.getHeaders());
-		headerMap.put("contentType", "application/json");
+		headerMap.remove("contentType");
```

Comment:
+1

Reply:
Any explanation "why", please?
Is this really how Open Source works?

Either way I would suggest to consider `MessageBuilder` API instead of these manual manipulations, including `new HashMap<>(message.getHeaders())` above.

Thanks